react/components/Spinner/index.tsx

```diff
@@ -1,28 +1,37 @@
-import React from 'react'
+import React, { FC } from 'react'
 import PropTypes from 'prop-types'
+import classNames from 'classnames'
 
 import { baseClassname } from '../icon/utils'
 
 const radius = 40
 const circ = 2 * radius * Math.PI
 
-class Spinner extends React.Component {
-  render() {
-    const { color, size, block } = this.props
+const propTypes = {
+  /** Color of the spinner */
+  color: PropTypes.string,
+  /** Size (diameter) of the spinner */
+  size: PropTypes.number,
+  /** Sets the display to block */
+  block: PropTypes.bool,
+}
+
+type Props = PropTypes.InferProps<typeof propTypes>
 
-    return (
-      <svg
-        className={`${baseClassname('spinner')} ${
-          !color ? 'c-action-primary' : ''
-        } ${block ? 'db' : ''}`}
-        xmlns="http://www.w3.org/2000/svg"
-        viewBox="0 0 100 100"
-        preserveAspectRatio="xMidYMid"
-        height={size}
-        width={size}
-      >
-        <style>
-          {`
+const Spinner: FC<Props> = ({ color, size, block }: Props) => (
```

Comment:
nit: I think that you could omit the props definition in brackets `const Spinner: FC<Props> = ({ color, size, block }) `

Reply:
Oops, sorry :sweat_smile: This was something that I was testing

Reply:
Resolved